app/src/main/java/io/github/trojan_gfw/igniter/servers/fragment/ServerListAdapter.java

```diff
@@ -22,18 +23,24 @@
 import io.github.trojan_gfw.igniter.TrojanConfig;
 import io.github.trojan_gfw.igniter.servers.ItemVerticalMoveCallback;
 import io.github.trojan_gfw.igniter.servers.data.TrojanConfigWrapper;
+import io.github.trojan_gfw.igniter.servers.data.ServerListDataManager;
 
 public class ServerListAdapter extends RecyclerView.Adapter<ViewHolder> implements ItemVerticalMoveCallback.ItemTouchHelperContract {
+    private final static String TAG = "ServerListAdapter";
+
     private final LayoutInflater mInflater;
     private final List<TrojanConfigWrapper> mData;
+    private final List<Float> mPingDelayTimeView;
```

Comment:
As you can see, I defined a class `TrojanConfigWrapper`. It is used for adding non-persistent, extra attributes for `TrojanConfig`. You can consider place the ping attribute inside `TrojanConfigWrapper` instead of declaring another list of ping. Just a suggestion.

Reply:
Good suggestion. I am working on it.